### PR TITLE
Add documentation on dependency management

### PR DESCRIPTION
**Pull request by danielpieczko**

### CODE REVIEW

File: doc/dependency_management.rst [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+XCommon CMake provides a dependency management solution which can fetch modules to be used in an application.
+These modules are cloned from git repositories and placed in the root of the sandbox.
+
+Starting from the application's CMakeLists.txt, the ``APP_DEPENDENT_MODULES`` variable defines the immediate
+dependencies of the application. When each ``lib_build_info.cmake`` file is included for each dependency, their
+``LIB_DEPENDENT_MODULES`` variables define the sub-dependency relationships. This builds up a tree which is
+traversed depth-first to populate the sandbox.
+
+As an example, suppose that an application's ``CMakLists.txt`` contains ``set(APP_DEPENDENT_MODULES lib_mod0 lib_mod1)``
```
Comment (xross) [inline doc/dependency_management.rst:12]: Typo: CMakLists.txt

File: doc/dependency_management.rst [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+a table for the location from which the module was cloned, and the branch/tag and git commit hash for the current checked
+out changeset. If any columns are not applicable, they will contain a hyphen.
+
+Dependency Location
+^^^^^^^^^^^^^^^^^^^
+
+By default, the location of dependent modules is the root of the sandbox as defined by the ``XMOS_SANDBOX_DIR``
+variable. This can be overridden on a per-dependency basis by setting a variable for each non-default dependency
+location.
+
```
Comment (xross) [inline doc/dependency_management.rst:57]: Add a ..note:: that this is the recommend behaviour

File: doc/dependency_management.rst (hunk 1 above)
Comment (danielpieczko) [reply doc/dependency_management.rst]: Fixed.

File: doc/dependency_management.rst (hunk 2 above)
Comment (danielpieczko) [reply doc/dependency_management.rst:57]: Done - this note directive is useful, I hadn't seen it before.

**Review by xross — APPROVED** [review]

**Pull request merged**